docs/examples/201/servicebus-create-queue/main.bicep

```diff
@@ -0,0 +1,32 @@
+param serviceBusNamespaceName string
+param serviceBusQueueName string
+param location string = resourceGroup().location
+
+resource serviceBusNamespace 'Microsoft.ServiceBus/namespaces@2017-04-01' = {
+  name: serviceBusNamespaceName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {}
+}
+
+resource serviceBusQueue 'Microsoft.ServiceBus/namespaces/queues@2017-04-01' = {
+  name: format('{0}/{1}', serviceBusNamespaceName, serviceBusQueueName)
+  properties: {
+    lockDuration: 'PT5M'
+    maxSizeInMegabytes: 1024
+    requiresDuplicateDetection: false
+    requiresSession: false
+    defaultMessageTimeToLive: 'P10675199DT2H48M5.4775807S'
+    deadLetteringOnMessageExpiration: false
+    duplicateDetectionHistoryTimeWindow: 'PT10M'
+    maxDeliveryCount: 10
+    autoDeleteOnIdle: 'P10675199DT2H48M5.4775807S'
+    enablePartitioning: false
+    enableExpress: false
+  }
+  dependsOn: [
```

Comment:
can get rid of lines 29-31 once the above is done. The dependsOn will be automatically added in the generated template

Reply:
Interesting. If I keep the format syntax in the example above the dependson property is not automatically added.

Reply:
when you use `format()` with the resource reference (`serviceBusNamespace.name`)?

Reply:
You're right, using `serviceBusNamespace.name` it generates the dependency.